lib/recording/RecordingManager.js

```diff
@@ -50,133 +50,83 @@ class RecordingManager {
-        return Q.resolve().then(()=> {
-            function makeFlavorDirectories(recordingEntrySession) {
-                logger.debug(`[${entryObj.entryId}-${entryObj.recordedEntryId}] About to create directories ${recordingEntrySession.flavors}`);
-                let creatingDirectoriesPromise = _.map(recordingEntrySession.flavors, (flavorId)=> {
-                    let flavorPath = path.join(recordingEntrySession.recordingSessionPath, flavorId)
-                    return qio.makeDirectory(flavorPath)
-                });
-                return Q.allSettled(creatingDirectoriesPromise)
-                    .then((results)=> {
-                        _.each(results, (promise)=> {
-                            if (promise.state === 'rejected') {
-                                if (promise.reason && promise.reason.code === 'EEXIST') {
-                                    logger.debug(`[${entryObj.entryId}-${entryObj.recordedEntryId}] path ${promise.reason.path} already exist`);
-                                }
-                                else {
-                                    logger.error(`[${entryObj.entryId}-${entryObj.recordedEntryId}] Failed to make directory ${ErrorUtils.error2string(promise.reason)}`);
-                                }
-                            }
-                        })
-                    })
-            }
-            function getRecordingEntrySession(entryObj) {
-                let entryId = entryObj.entryId;
-                if (!entryObj.recordedEntryId) {
-                    entryObj.recordedEntryId =  unknownEntry;
-                    logger.warn(`[${entryObj.entryId}] Entry has no recordedEntryId`);
-                }
-                if (!this.recordingList[entryId]) {
-                    logger.info(`[${entryObj.entryId}-${entryObj.recordedEntryId}] Create new recording session`);
-                    let recordingEntrySession = new RecordingEntrySession(entryObj);
+    getRecordingEntrySession(entryObj, mp4FilesQueue = null) {
+        let entryId = entryObj.entryId;
+        if (!entryObj.recordedEntryId) {
+            entryObj.recordedEntryId = unknownEntry;
+            logger.warn(`[${entryObj.entryId}] Entry has no recordedEntryId`);
+        }
+        if (!this.recordingList[entryId]) {
+            logger.info(`[${entryObj.entryId}-${entryObj.recordedEntryId}] Create new recording session`);
+            let recordingEntrySession = new RecordingEntrySession(entryObj);
+            this.recordingList[entryId] = recordingEntrySession;
+            return Q.resolve({session:recordingEntrySession, started:false});
+        }
+        else if (this.recordingList[entryId].recordedEntryId != entryObj.recordedEntryId) {
+            logger.info(`[${entryObj.entryId}-${entryObj.recordedEntryId}] New recorded entryId has been generated. Delete existing session and start new one`);
+            return this.stopRecording(entryObj.entryId)
```

Comment:
it doesn't sound logic to me that the function that is called getRecordingEntrySession is doing stopRecording